paas/install_cli.go

```diff
@@ -1,35 +1,115 @@
 package paas
 
 import (
+	"encoding/json"
 	"fmt"
+	"io/ioutil"
 	"os"
 	"path/filepath"
+	"regexp"
 	"runtime"
+	"strings"
 
 	"github.com/fuseml/fuseml/cli/deployments"
 	"github.com/fuseml/fuseml/cli/helpers"
 	"github.com/fuseml/fuseml/cli/paas/ui"
+	"github.com/fuseml/fuseml/cli/paas/version"
 	"github.com/pkg/errors"
 )
 
 const (
-	coreClientDownloadURL = "https://github.com/fuseml/fuseml-core/releases/latest/download"
+	coreAPIURL            = "https://api.github.com/repos/fuseml/fuseml-core/releases"
+	coreClientDownloadURL = "https://github.com/fuseml/fuseml-core/releases"
 	coreClientName        = "fuseml"
 )
 
+// structure covering tiny part of github release json information
+type Release struct {
+	URL     string `json:"html_url,omitempty"`
+	TagName string `json:"tag_name,omitempty"`
+}
+
+// Find out latest applicable version of the core client for the installer;
+// installer version provided as argument
+func latestClientForInstaller(version string) (string, error) {
+	tmpDir, err := ioutil.TempDir("", "fuseml-core-releases")
+	if err != nil {
+		return "", errors.Wrap(err, "can't create temp directory "+tmpDir)
+	}
+	defer os.RemoveAll(tmpDir)
+
+	releasesFile := "releases.json"
+	if err := helpers.DownloadFile(coreAPIURL, releasesFile, tmpDir); err != nil {
+		return "", err
+	}
+
+	data, err := os.ReadFile(filepath.Join(tmpDir, releasesFile))
+	if err != nil {
+		return "", errors.Wrap(err, "failed to read description file")
+	}
```

Comment:
It feels like you go to a lot of trouble here just to read the releases JSON manifest. You don't actually need to download it and then read it from disk, you can just read it from the HTTP response body.

Reply:
You are right, of course. I just started with existing download method and then just read a file.

I admit it's unnecessary, though I would not agree it's "a lot of trouble". I guess I could change it though